### PR TITLE
refactor mappers: heatpump

### PR DESCRIPTION
**Pull request by AshrafAlmohagryDeltares**

### CODE REVIEW

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class EsdlAssetHeatPumpMapper(EsdlMapperAbstract):
+    """Class to map an ESDL asset to a heatpump entity class."""
+
+    def to_esdl(self, entity: HeatPump) -> EsdlAssetObject:
+        """Map a HeatPump entity to an EsdlAsset."""
+        raise NotImplementedError("EsdlAssetHeatPumpMapper.to_esdl()")
+
+    def to_entity(self, esdl_asset: EsdlAssetObject) -> AssetAbstract:
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py:30]: This mapper will require a get COP from the model, but I will add it in the branch of the HP.

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 2]
```diff
@@ -60,11 +63,25 @@ def to_entity(model: EsdlAssetObject) -> AssetAbstract:
 
         :return: Entity object of type AssetAbstract.
         """
-        if not type(model.esdl_asset) in CONVERSION_DICT:
+        if (
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:66]: First resolve https://github.com/Project-OMOTES/simulator-core/pull/194

**Review by vanmeerkerk — COMMENTED** [review]
Maybe next time we should make branches (a1, a2 etc) on the branch (a) so that these branches (a1, a2, etc) can then request a review to be merged on branch (a). 

In that way we can make sure the basic implementation is correct and won't have to fix dependies in different branches.

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py (hunk 1 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py:30]: what are COP and HP?

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py (hunk 2 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py:66]: completed.

**Review by vanmeerkerk — DISMISSED** [review]
Changes have been adressed

File: src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py (hunk 1 above)
Comment (vanmeerkerk) [reply src/omotes_simulator_core/adapter/transforms/esdl_asset_mappers/EsdlAssetHeatPumpMapper.py]: Coefficient of Performance (COP) of the Heat Pump (HP) these must be added in the HP branch.

**Review by MichielTukker — APPROVED** [review]

**Review by MichielTukker — DISMISSED** [review]

**Review by MichielTukker — APPROVED** [review]

**Comment by MichielTukker** [comment]
Overriding branch protections,  as I've only added an mypy ignore

**Pull request merged**